pkg/monitor/monitor.go

```diff
@@ -206,10 +206,11 @@ func (m *Monitor) driverMonitor() {
 			}
 			nodes = volume.RemoveDuplicateOfflineNodes(nodes)
 			for _, node := range nodes {
-				// Check if nodes are reported online by the storage driver
-				// If not online, look at all the pods on that node
+				// Check if nodes are reported as offline or degraded by the storage driver
+				// If offline or degraded, look at all the pods on that node
 				// For any Running pod on that node using volume by the driver, kill the pod
-				if node.Status != volume.NodeOnline {
+				// Degraded nodes are not considered offline and pods are not deleted from them.
+				if node.Status == volume.NodeOffline || node.Status == volume.NodeDegraded {
```

Comment:
During exponential backoff we check for node condition again here 
https://github.com/libopenstorage/stork/blob/master/pkg/monitor/monitor.go#:~:text=if%20n.Status%20!%3D%20volume.NodeOnline%20%7B
This condition should be updated too.